boards/arm/stm32f723e_disco/stm32f723e_disco_defconfig

```diff
@@ -12,6 +12,9 @@ CONFIG_SERIAL=y
 CONFIG_CONSOLE=y
 CONFIG_UART_CONSOLE=y
 
+# enable I2C
+CONFIG_I2C=y
```

Comment:
Should be 'n' by default.

Reply:
@erwango We enable buses on board level when there is a peripheral (e.g. WM8994 or touch panel in this case) that needs them, aren't we?

Reply:
@ydamigos : we should configure them, and this is what is done here. But activation should always be done in application and not default.
Aim is that several developers on the same board know what is the base components for the board so they only enable what they need. But they don't have to bother disabling what other developers may have enabled meanwhile.

Reply:
@erwango We need to clean up board _defconfig files then.

Reply:
@ydamigos : yes, there is some work ahead! I raised https://github.com/zephyrproject-rtos/zephyr/issues/7151 earlier, waiting https://github.com/zephyrproject-rtos/zephyr/pull/7176 to be reviewed and used a reference (which was locked waiting for new SDK version and which I hope should get some feedbacks soon)